### PR TITLE
Add text and validation to block PO boxes for Mimeo orders

### PR DESCRIPTION
**Pull request by aoby**
Added `(No PO Boxes)` text to the street address field:

image

And validation to prevent it:

image

image

### CODE REVIEW

**Comment by aoby** [comment]
Good suggestion! Will update

**Comment by aoby** [comment]
Updated with Geocoder. PTAL

File: dashboard/app/models/pd/workshop_material_order.rb [hunk 1]
```diff
@@ -69,8 +69,13 @@ def valid_address?
       found = Geocoder.search(full_address)
       if found.empty?
         errors.add(:base, 'Address could not be verified. Please double-check.')
-      elsif found.first.postal_code != zip_code
-        errors.add(:zip_code, "doesn't match the address. Did you mean #{found.first.postal_code}?")
+      else
+        if found.first.postal_code != zip_code
+          errors.add(:zip_code, "doesn't match the address. Did you mean #{found.first.postal_code}?")
+        end
+        unless found.first.address_components.any? {|c| c['types'].include? 'street_number'}
```
Comment (Hamms) [inline dashboard/app/models/pd/workshop_material_order.rb]: I believe you can just do `found.first.street_number`:

```
@irb(main):005:0> Geocoder.search("PO Box 123").first.street_number
=> nil
@irb(main):006:0> Geocoder.search("1501 4th Ave, Seattle, WA").first.street_number
=> "1501"
```

Comment (aoby) [reply dashboard/app/models/pd/workshop_material_order.rb]: Oh nice. That's much simpler! Thanks.

**Pull request merged**